### PR TITLE
Kubernetes step parsing and command line options

### PR DESCRIPTION
**Pull request by hylje**
Included in this PR:
===

1. Parse Kubernetes `resource` structure from YAML when using `vh execution run`, if provided.
2. Parse Kubernetes resource claims directly as `vh execution run` command line arguments

`vh pipeline run` does not have execution runtime option override support for now.

### CODE REVIEW

File: tests/commands/execution/test_run.py [hunk 1]
```diff
@@ -252,3 +261,133 @@ def test_remote(run_test_setup, tmpdir):
...
+    run_test_setup.args.append("--k8s-cpu-min=1")
+    run_test_setup.args.append("--k8s-memory-min=2")
+    run_test_setup.args.append("--k8s-cpu-max=3")
+    run_test_setup.args.append("--k8s-memory-max=4")
+    run_test_setup.args.append("--k8s-device=nvidia.com/gpu=1")
+    run_test_setup.run()
+
+    assert run_test_setup.run_api_mock.last_create_execution_payload["runtime_config"] == {
+        "kubernetes": {
+            'containers': {
```
Comment (ruksi) [inline tests/commands/execution/test_run.py:276]: mixing of the `"` and `'` hurts my souls a bit but doesn't really matter :ghost:

File: tests/commands/execution/test_run.py [hunk 2]
```diff
@@ -252,3 +261,133 @@ def test_remote(run_test_setup, tmpdir):
...
+                'workload': {
+                    'resources': {
+                        'requests': {
+                            'cpu': 1.0,
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (ruksi) [inline tests/commands/execution/test_run.py:313]: the rest are not sent in the payload as we except that the API fills them from the YAML :+1:

File: tests/commands/execution/test_run.py [hunk 3]
```diff
@@ -252,3 +261,133 @@ def test_remote(run_test_setup, tmpdir):
...
+                        'limits': {
+                            'cpu': 2.0,
+                            'memory': 4,
+                            'devices': {}
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (ruksi) [inline tests/commands/execution/test_run.py:393]: not sure I understand this, why are the non-gpu resources sent in this case?

**Review by ruksi — CHANGES_REQUESTED** [review]
lgtm, but a _question_

File: tests/commands/execution/test_run.py (hunk 3 above)
Comment (hylje) [reply tests/commands/execution/test_run.py:393]: Resources from YAML are parsed and sent if available (see the `run_test_setup_kube` fixture), and command line arguments can override them further. `--k8s-device-none` overrides just the devices dict into an explicit empty dict, while other resources are not overridden.

In the above partial case, the YAML doesn't define any resources so we won't send them either

Comment (ruksi) [reply tests/commands/execution/test_run.py:393]: but if we send YAML defined resources, wouldn't that mix up e.g. future copying? (we don't know know which of the passed parameters are API overrides vs. YAML overrides)

assuming that API doesn't compare passed resources to the YAML it knows 🤔

Comment (ruksi) [reply tests/commands/execution/test_run.py:393]: but I guess it's fine if it work like the wen UI, could double check it

Comment (hylje) [reply tests/commands/execution/test_run.py:393]: I don't think executions keep track of where the resulting configuration comes from, it's all put together and saved for later use. Copying doesn't refresh changes from yaml Steps, whether or not they were also supplied through the API or not.

I think the principle that front-end resolves the YAML itself and submits from it is alright, we already do it in CLI for parameters, inputs and environment variables.

I'm not against just leaving it for the API backend to work out, but I think that's its own project.

Comment (ruksi) [reply tests/commands/execution/test_run.py:393]: > I think the principle that front-end resolves the YAML itself and submits from it is alright

yeah, I also feel that it's all-OK :+1: 

> I'm not against just leaving it for the API backend to work out, but I think that's its own project

yeah, definitely a separate thing if we feel that it's needed

**Review by ruksi — APPROVED** [review]
is good

**Pull request merged**